src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js

```diff
@@ -1,18 +1,111 @@
 
 export default class geFormWidgetTokenizeCard extends LightningElement {
 
     CUSTOM_LABELS = GeLabelService.CUSTOM_LABELS;
     
-    @track token = 'token_placeholder';
+    tokenizeCardPageUrl = '/apex/GE_TokenizeCard';
+    @track domain;
+    @track visualforceOrigin;
+    @track isLoading = true;
 
     get tokenizeCardHeader() {
         return GeLabelService.format(
             this.CUSTOM_LABELS.geHeaderPaymentServices,
             [this.CUSTOM_LABELS.commonPaymentServices]);
     }
-    // TODO: public method for returning widget data
-    // TODO: method for calling tokenize method in iframe and retrieving the token
+
+    async connectedCallback() {
+        this.domain = await getOrgDomain();
+        this.visualforceOrigin = this.buildVisualforceOriginUrl(this.domain);
+    }
+
+    renderedCallback() {
+        this.registerPostMessageListener();
+    }
+
+    /*******************************************************************************
+    * @description Builds the visualforce origin url that we need in order to
+    * make sure we're only listening for messages from the correct source in the
+    * registerPostMessageListener method.
+    */
+    buildVisualforceOriginUrl(domain) {
+        let url = `https://${domain}--c.visualforce.com`
+        if (TemplateBuilderService.namespaceWrapper) {
+            const currentNamespace = TemplateBuilderService.namespaceWrapper.currentNamespace;
+
+            if (currentNamespace) {
+                url = url.replace('--c', `--${currentNamespace}`);
+            }
+        }
+
+        return url;
+    }
+
+    /*******************************************************************************
+    * @description Method listens for a message from the visualforce iframe.
+    * Rejects any messages from an unknown origin.
+    */
+    registerPostMessageListener() {
+        let component = this;
+
+        window.onmessage = async function (event) {
+            if (event && event.origin !== component.visualforceOrigin) {
+                // Reject any messages from an unexpected origin
+                return;
+            } else {
+                const message = JSON.parse(event.data);
+                component.handleMessage(message);
+            }
+        }
+    }
+
+    /*******************************************************************************
+    * @description Method handles messages received from iframed visualforce page.
+    *
+    * @param {object} message: Message received from iframe
+    */
+    async handleMessage(message) {
+        if (message.error) {
+            // Error with tokenization
+            let error = JSON.stringify(message.error);
+            handleError(error);
+        } else if (message.token) {
+            // TODO: Start - Remove later
+            // Make purchase call... for dev only
+            try {
+                let purchaseCallResponse = await makePurchaseCall({ token: message.token });
```

Comment:
Should the purchase callout be made inside this "Tokenize" component?  Maybe this component should be encapsulated and solely responsible for tokenizing and returning a value (the token) and then the FormRenderer should make the callout within (or before) its "save" flow...  Thoughts?

Reply:
I think this is ok for SGE, but we need to make sure that `makePurchaseCall` calls something that BGE can use later on.

Alternatively, If we want the DI record to be populated and ready for import at the moment we make the purchase call, we should return the token to the formRenderer and let it take over from there.

I'm leaning slightly towards the second option, where the formRenderer receives the token and makes the purchase call.

Reply:
I am leaning towards Ken's comment here, based on the component hierarchy we currently have, it might be better if we let FormRenderer be that one point for communicating with the back-end, It also just makes for an easily maintainable codebase.

Reply:
Would definitely not want to keep this here. Basically just left in here as an example to be removed later (which could be now?). I was thinking this would basically just receive the token from the iframe and saved locally in the component for the form renderer to later retrieve. Something along those lines.